### ISSUE TITLE
More barcode detection speed improvement is required

### ISSUE THREAD

**Issue by monajafi** [googlesamples/android-vision#188]
While the vision api is very good at detecting barcodes, but still is weak in detection speed compared to XZing or Scandit library. It's also weaker in detecting damaged barcodes 
I am developing an mPos app for retail which requires detection speed that can be well achieved by just Scandit api. vision api is not satisfactory in retail barcode scanning.
Is there a detection speed improvement in vision api roadmap? when google play services will be updated?

**Comment by plugie** [comment]
I have the same issue. The engine can't detect barcode in some mobile phones. But Zxing is working well for those mobile phones

**Comment by monajafi** [comment]
I've compared XZing and vision barcode more again here are differences:
1-XZing: 
Pros:
  - detection speed is more if it's able to detect barcode both at legacy and new devices
Cons:
- can not detect at different orientations( vertical or inclined will not be detected)
- damaged or some small barcodes was not detected.

2- Vision Api
Pros:
      - number of successful tough barcode (small and so on) detection is more
      - different orientaions was detected .
     - it can be combined with text api to read damaged barcode numbers
Cons:
     - detection speed is little less specially in legacy devices. in newer devices like S7 speed is good 
      but Scandit api still is detecting too much faster regardless of device type which makes it like industrial  barcode detectors

**Comment by pchx** [comment]
Please describe the environment in which you are comparing the two APIs.

What kind of devices, camera frame sizes, barcode formats are you testing with? Do any specific combinations lead to performance problems? How much slower is the Vision API in these configurations?

Finally, note that higher overall latency is the price for detecting damaged or vertical barcodes. If you don't require that functionality and that latency is not acceptable for your use case, then perhaps ZXing is a more appropriate choice.

**Comment by monajafi** [comment]
I've tested using galaxy S4 mini ,galaxy S5 and S6 edge plus,S7 in many frame combinations mostly for EAN13 which was printed on Retail products. detection is not instant most of time there is 1 second or 2 second delay before detection, which makes vision api inappropriate for retail barcode scanning because regular barcode scanners from brands like DataLogic detect instantly without any delay.
I've tested every available android barcode scanner I've found just Scandit api can compete with regular and industrial barcode scanners. 
I wish Google vision api team work on the issue in next google play services updates.

**Comment by suryamegham1729** [comment]
Yes, Auto focusing is taking too long time compare to other scan engines

**Comment by pulmuone** [comment]
google vision barcode is faster than zxing.
